File: ontrack-extension-vault/src/main/java/net/nemerosa/ontrack/extension/vault/VaultConfigProperties.java
package net.nemerosa.ontrack.extension.vault;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Data
@Component
@ConfigurationProperties(prefix = "ontrack.config.vault")
public class VaultConfigProperties {

    /**
     * URI to the Vault end point
     */
    private String uri = "http://localhost:8200";

    /**
     * Token authentication
     */
    private String token = "test";

    /**
     * Key prefix
     */
    private String prefix = "ontrack/secrets/key";

}

File: ontrack-extension-vault/src/main/java/net/nemerosa/ontrack/extension/vault/VaultConfigProperties.kt
package net.nemerosa.ontrack.extension.vault

import net.nemerosa.ontrack.model.annotations.APIDescription
import net.nemerosa.ontrack.model.annotations.APIName
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties(prefix = "ontrack.config.vault")
@APIName("Vault configuration")
@APIDescription("Ontrack can be configured to use Vault to store the encryption keys.")
class VaultConfigProperties {

    @APIDescription("URI to the Vault end point")
    var uri = "http://localhost:8200"

    @APIDescription("Token for the authentication")
    var token = "test"

    @APIDescription("Prefix to be used to store the keys")
    var prefix = "ontrack/keys"
}